tools/nnyeah/tests/utils/Compiler.cs

```diff
@@ -75,13 +76,38 @@ static string PlatformLibPath (PlatformName platformName, string libName)
 				_ => throw new NotImplementedException (),
 			};
 
-		static string XamarinLibName (PlatformName platformName) =>
+		public static string XamarinLibName (PlatformName platformName) =>
 			platformName switch {
 				PlatformName.macOS => "Xamarin.Mac",
 				PlatformName.iOS => "Xamarin.iOS",
 				PlatformName.tvOS => "Xamarin.TVOS",
 				PlatformName.watchOS => "Xamarin.WatchOS",
 				_ => throw new NotImplementedException (),
 			};
+
+		public static string XamarinPlatformLibraryPath (PlatformName platformName) =>
+			Path.Combine (XamarinPlatformLibDirectory (platformName), $"{XamarinLibName (platformName)}.dll");
+
+		public static string MicrosoftPlatformLibDirectory (PlatformName platformName) =>
+			platformName switch {
+				PlatformName.macOS => Path.Combine (buildDirectory, "Microsoft.macOS.Runtime.osx-arm64/runtimes/osx-arm64/lib/net6.0"),
+				PlatformName.iOS => Path.Combine (buildDirectory, "Microsoft.iOS.Runtime.ios-arm/runtimes/ios-arm/lib/net6.0"),
+				PlatformName.tvOS => Path.Combine (buildDirectory, "Microsoft.tvOS.Runtime.tvossimulator-x64/runtimes/tvossimulator-x64/lib/net6.0"),
+				PlatformName.watchOS => throw new NotImplementedException (),
+				_ => throw new NotImplementedException (),
```

Comment:
Would it be helpful to throw in the name of the platform?
```suggestion
				_ => throw new NotImplementedException (nameof (platformName)),
```